java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -287,7 +288,7 @@ private boolean hasConstraintForDomain(LearnedConstraint lc, Class<? extends Con
         // but this is necessary so path is returned in #addEdge and stays in the worklist in #run
         return ImmutableSet.of(ImmutableList.of());
       }
-
+      Preconditions.checkState(isMethodInvocationNode(edge.parent), "Yield present on edge where parent is not MIT");
```

Comment:
I am not really sure this is the place where we actually want to check this and I don't really see what's the risk you address with this invariant.